### ISSUE TITLE
Restconf in YDK should support HTTPS

### ISSUE THREAD

**Issue by ghost** [CiscoDevNet/ydk-gen#917]

**Comment by ygorelik** [comment]
Added partial support for HTTPS protocol for Restconf. The 'partial' means that YDK is capable communicate with Restconf server over secure protocol, provides data encryption, checks CA certificate, but the peer and host name verification are currently permanently disabled (hard-coded). This limitation should be addressed in future YDK releases.

In the application the user must explicitly specify HTTPS protocol in the host address. The following example shows, how the RestconfServiceProvider is used to read names of all interfaces from secure Restconf server:
```
from ydk.services import CRUDService
from ydk.providers import RestconfServiceProvider
from ydk.types import EncodingFormat
from ydk.path import Repository
from ydk.models.openconfig import openconfig_interfaces

if __name__ == '__main__':
    repo = Repository('/Users/ygorelik/.ydk/sbx-iosxr-mgmt.cisco.com')
    provider = RestconfServiceProvider(
        repo,
        'https://ios-xe-mgmt.cisco.com',   # Add 'https://' prefix to the host name or address
        'developer',
        'C1sco12345',
        9443,    # HTTPS port
        EncodingFormat.JSON)

    interfaces = openconfig_interfaces.Interfaces()

    crud = CRUDService()
    all_interfaces = crud.read(provider, interfaces)

    for intf in all_interfaces.interface:
        print(intf.name)
```

**Comment by ygorelik** [comment]
Resolved in release 0.8.5
